utils/build/webpack.base.js

```diff
@@ -67,7 +67,7 @@ found at http://www.apache.org/licenses/LICENSE-2.0
         use: [
           {
             loader: 'babel-loader',
-            options: babelOptions
+            options: { ...babelOptions, envName: 'production' }
```

Comment:
Turns out setting `mode: production` in webpack doesn't set the global `NODE_ENV` variable to `production`. This change tells the babel-loader to use our `production` environment to remove the `data-test-*` attributes.